modin/experimental/cloud/test/test_cloud.py

```diff
@@ -42,10 +40,14 @@ def test_update_conda_requirements(setup_commands_source):
             Provider(name="aws"), add_conda_packages=["scikit-learn>=0.23"]
         )
 
-    setup_commands_result = ray_cluster._update_conda_requirements(
-        setup_commands_source
-    )
+    with mock.patch(
+        "sys.version_info", namedtuple("mock", version_info.keys())(**version_info)
```

Comment:
That is dangerous... dictionaries don't have defined order of keys and items, so, even if now it should probably be working as the algorithm for iterating over `.keys()` and `**version_info` is same this is not guaranteed.

I propose to combine declaration of mock version info like this (untested):
```python
fake_version = namedtuple("FakeVersion", "major minor micro")(7, 12, 45)
with mock.patch("sys.version_info", fake_version):
    # do things
assert f"python>={fake_version.major}.{fake_version.minor}"
assert f"python<={fake_version.major}.{fake_version.minor}.{fake_version.micro}"
```

Reply:
Good point.
Done in the way.